test/powershell/Language/Scripting/CommandErrorHandling.Tests.ps1

```diff
@@ -0,0 +1,470 @@
+        }
+
+        It "'-ErrorAction Stop' stops execution with 'ActionPreference.<ErrorAction>' for error action" -TestCases @(
+            @{ ErrorAction = 'Continue' }
+            @{ ErrorAction = 'Ignore' }
+            @{ ErrorAction = 'SilentlyContinue' }
+        ) {
+            param ($ErrorAction)
+
+            $failure = $null
+
+            try {
+                RunCommand -Command 'ErrorActionStop' -ErrorAction $ErrorAction
+            } catch {
+                $failure = $_
+            }
+
+            $failure | Should -Not -BeNullOrEmpty
+            $failure.Exception | Should -BeOfType 'System.Management.Automation.MethodInvocationException'
+            $failure.Exception.InnerException | Should -BeOfType 'System.Management.Automation.ActionPreferenceStopException'
+            $failure.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should -BeExactly 'CommandNotFoundException,Microsoft.PowerShell.Commands.GetCommandCommand'
+
+            $pwsh.Streams.Verbose.Count | Should -BeExactly 0
+        }
+
+        It "'throw' statement stops execution with 'ActionPreference.Continue' (the default error action)" {
+            $failure = $null
+
+            try {
+                RunCommand -Command 'ThrowException' -ErrorAction 'Continue'
+            } catch {
+                $failure = $_
+            }
+
+            $failure | Should -Not -BeNullOrEmpty
+            $failure.Exception | Should -BeOfType 'System.Management.Automation.MethodInvocationException'
+            $failure.Exception.InnerException | Should -BeOfType 'System.Management.Automation.RuntimeException'
+            $failure.Exception.InnerException.Message | Should -BeExactly 'throw-exception'
+
+            $pwsh.Streams.Verbose.Count | Should -BeExactly 0
+        }
+
+        <# The 'throw' statement is special, in that it can be suppressed by '-ErrorAction SilentlyContinue/Ignore' #>
+        It "'throw' statement continues execution with 'ActionPreference.<ErrorAction>'" -TestCases @(
+            @{ ErrorAction = 'Ignore' }
+            @{ ErrorAction = 'SilentlyContinue' }
+        ) {
+            param ($ErrorAction)
+
+            $failure = $null
+
+            try {
+                RunCommand -Command 'ThrowException' -ErrorAction $ErrorAction
+            } catch {
+                $failure = $_
+            }
+
+            $failure | Should -BeNullOrEmpty
+            $pwsh.Streams.Verbose.Count | Should -BeExactly 1
+            $pwsh.Streams.Verbose | Should -BeExactly 'verbose-message'
+
+            ## The suppressed 'throw' exception is not written to the error stream, not sure why but it's the current behavior.
+            $pwsh.Streams.Error.Count | Should -BeExactly 0
+
+            ## The suppressed 'throw' exception is kept in '$Error'
+            $err = GetLastError
+            $err | Should -Not -BeNullOrEmpty
+            $err.FullyQualifiedErrorId | Should -BeExactly 'throw-exception'
```

Comment:
What is a behavior in previous tests? Maybe add such checks there too?

As result we could combine all these tests in one with TestCases - it could be more readable to see one config table.

Reply:
The previous tests are dealing with terminating errors, which are surfaced as exceptions. Yeah, the exception errors should be kept in `$Error` as well, but that was not the point of those tests.

Please be noted that those tests are by no means covering every aspect of the error handling behaviors. I tried to cover most of the subtle behaviors -- some are not well known and may even be a little confusing to some extent. This is because these subtle behaviors may not be well tested, and adding those new tests may help me from introducing regression without knowing. For the more obvious/common behaviors, I hope they are already covered by existing tests.